test/e2e/specs/editor/various/preview.spec.js

```diff
@@ -0,0 +1,362 @@
+	test.beforeEach( async ( { admin, previewUtils } ) => {
+		await admin.createNewPost();
+		await previewUtils.toggleCustomFieldsOption( true );
+	} );
+
+	test.afterEach( async ( { previewUtils } ) => {
+		await previewUtils.toggleCustomFieldsOption( false );
+	} );
+
+	// Catch regressions of https://github.com/WordPress/gutenberg/issues/12617
+	test( 'displays edits to the post title and content in the preview', async ( {
+		editor,
+		page,
+		previewUtils,
+	} ) => {
+		const editorPage = page;
+
+		// Add an initial title and content.
+		await editorPage.type( 'role=textbox[name="Add title"i]', 'title 1' );
+		await editorPage.type(
+			'role=button[name="Add default block"i]',
+			'content 1'
+		);
+
+		// Publish the post and then close the publish panel.
+		await editor.publishPost();
+
+		// Close the panel.
+		await page.locator( 'role=button[name="Close panel"]' ).click();
+
+		// Open the preview page.
+		const previewPage = await editor.openPreviewPage();
+		await previewPage.waitForSelector( 'role=heading[level=1]' );
+
+		// Check the title and preview match.
+		const previewTitle = previewPage.locator( 'role=heading[level=1]' );
+		expect( await previewTitle.textContent() ).toEqual( 'title 1' );
+		// No semantics we can grab here; it's just a <p> inside a <div> :')
+		const previewContent = previewPage.locator( '.entry-content p' );
+		expect( await previewContent.textContent() ).toEqual( 'content 1' );
+
+		// Return to editor and modify the title and content.
+		await editorPage.bringToFront();
+		await editorPage.click( 'role=textbox[name="Add title"i]' );
+		await editorPage.keyboard.press( 'End' );
+		await editorPage.keyboard.press( 'Backspace' );
+		await editorPage.keyboard.type( '2' );
+		await editorPage.click( 'role=document >> text="content 1"' );
+		await editorPage.keyboard.press( 'End' );
+		await editorPage.keyboard.press( 'Backspace' );
+		await editorPage.keyboard.type( '2' );
+
+		// Open the preview page.
+		await previewUtils.waitForPreviewNavigation( previewPage );
+
+		// Title in preview should match input.
+		expect( await previewTitle.textContent() ).toEqual( 'title 2' );
+
+		expect( await previewContent.textContent() ).toEqual( 'content 2' );
+
+		// Make sure the editor is active for the afterEach function.
+		await editorPage.bringToFront();
+	} );
+} );
+
+test.describe( 'Preview with private custom post type', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin( 'gutenberg-test-custom-post-types' );
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-custom-post-types'
+		);
+	} );
+
+	test( 'should not show the Preview Externally link', async ( {
+		admin,
+		page,
+	} ) => {
+		await admin.createNewPost( { postType: 'not_public' } );
+
+		// Type in the title filed.
+		await page.type( 'role=textbox[name="Add title"i]', 'aaaaa' );
```

Comment:
Could we use the `title` field in `createNewPost` here to simplify it?

```suggestion
		await admin.createNewPost( { postType: 'not_public', title: 'aaaaa' } );
```

Ditto to other places above too.